src/iocp.rs

```diff
@@ -163,6 +174,7 @@ impl CompletionPort {
         };
         match crate::cvt(ret) {
             Ok(_) => Ok(&mut list[..removed as usize]),
+            Err(e) if e.raw_os_error() == Some(WAIT_TIMEOUT as i32) => Ok(&mut list[..0]),
```

Comment:
I don't understand all the implications of things timing out at this level, but it feels like a more consistent API if users would handle timeouts that way most of the time anyway. So :+1: